### ISSUE TITLE
Question about the diag_mask in CTran.py file

### ISSUE THREAD

**Issue by iamstarlee** [QData/C-Tran#23]
> output = self.output_linear(label_embeddings) 
        diag_mask = torch.eye(output.size(1)).unsqueeze(0).repeat(output.size(0),1,1).cuda()
        output = (output*diag_mask).sum(-1)

I'm wondering what this diag_mask is playing in output value? Why should the output multiple a diagonal matrix? Your kind reply is very appreciated!

**Comment by jacklanchantin** [comment]
This is a method to obtain the logits for each individual class. The output is of dim LxL, by multiplying by diag_mask and summing, we get the Lx1 vector.

**Comment by iamstarlee** [comment]
Alright! Thank you very much!! you are so nice! your paper inspired me a lot, it is really a good job! I am trying to classify sewer defect using C-Tran, but the recall scores are not satisfactory, hope I can get solution the next second.
